calnex/cert/cert.go

```diff
@@ -0,0 +1,195 @@
+/*
+Copyright (c) Facebook, Inc. and its affiliates.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cert
+
+import (
+	"bytes"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"strings"
+	"time"
+)
+
+// This are the errors that can be raised by certificate parsing and validation
+var (
+	ErrBundleNil           = errors.New("bundle is nil")
+	ErrBundleNoCertForHost = errors.New("bundle has no certificate for supplied host")
+	ErrBundleNoCerts       = errors.New("bundle has no certificates")
+	ErrBundleNoPrivKey     = errors.New("bundle has no private key")
+	ErrCertExpired         = errors.New("certificate has expired")
+	ErrCertNotYetValid     = errors.New("certificate is not yet valid")
+	ErrFailedToParsePEM    = errors.New("failed to parse certificate PEM")
+	ErrMultiplePrivKeys    = errors.New("multiple private keys in PEM")
+	ErrOnlyRSA             = errors.New("only RSA private keys are supported")
+	ErrUnsupportedPEMBlock = errors.New("unsupported PEM block")
+)
```

Comment:
Does it need to be exposed? I think `errBundleNil` is good for your purposes :)

Reply:
It's possible a future user might want to check the error. If we're defining a error variable for each error we return we may as well provide the option to make use of them externally.

Reply:
Fair, we can leave them. But if you want you will definitely need to comment every one of them :)